libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -34,6 +36,74 @@ using namespace dev::solidity;
 
 using Scope = dev::solidity::assembly::Scope;
 
+CodeTransform::CodeTransform(
+	yul::AbstractAssembly& _assembly,
+	solidity::assembly::AsmAnalysisInfo& _analysisInfo,
+	Block const& _block,
+	solidity::AssemblyStack::Language _language,
+	bool _allowStackOpt,
+	bool _yul,
+	bool _evm15,
+	ExternalIdentifierAccess const& _identifierAccess,
+	bool _useNamedLabelsForFunctions,
+	int _stackAdjustment,
+	std::shared_ptr<Context> _context
+):
+	m_assembly(_assembly),
+	m_info(_analysisInfo),
+	m_language(_language),
+	m_allowStackOpt((_language == AssemblyStack::Language::StrictAssembly || _language == AssemblyStack::Language::Yul) && _allowStackOpt),
+	m_yul(_yul),
+	m_evm15(_evm15),
+	m_useNamedLabelsForFunctions(_useNamedLabelsForFunctions),
+	m_identifierAccess(_identifierAccess),
+	m_stackAdjustment(_stackAdjustment),
+	m_context(_context)
+{
+	if (m_allowStackOpt)
+		m_references = ReferencesCounter::countReferences(_block);
```

Comment:
Hm, I just realized that `ReferenceCounter` only works for disambiguated sources, i.e. for sources that do not have name clashes, because it counts based on the name alone. I fear we have to write another reference counter that uses scopes.